subprojects/workers/src/integTest/groovy/org/gradle/workers/internal/WorkerExecutorIntegrationTest.groovy

```diff
@@ -694,6 +694,110 @@ class WorkerExecutorIntegrationTest extends AbstractWorkerExecutorIntegrationTes
         isolationMode << ISOLATION_MODES
     }
 
+    def "workers which use project outputs do not break clean"() {
+        buildFile << """
+            plugins {
+                id 'java'
+                id 'test.worker-running-plugin'
+            }
+        """
+
+        file("settings.gradle") << "includeBuild 'included'"
+        file("included/settings.gradle") << "rootProject.name = 'included'"
+        file("included/build.gradle") << """
+            plugins {
+                id 'java-gradle-plugin'
+            }
+
+            gradlePlugin {
+                plugins {
+                    register("workerRunningPlugin") {
+                        id = "test.worker-running-plugin"
+                        implementationClass = "com.plugin.WorkerRunningPlugin"
+                    }
+                }
+            }
+        """
+        file("included/src/main/java/com/plugin/WorkerRunningPlugin.java") << """
+            package com.plugin;
+
+            import org.gradle.api.DefaultTask;
+            import org.gradle.api.NamedDomainObjectProvider;
+            import org.gradle.api.Plugin;
+            import org.gradle.api.Project;
+            import org.gradle.api.artifacts.Configuration;
+            import org.gradle.api.artifacts.ConfigurationContainer;
+            import org.gradle.api.file.FileCollection;
+            import org.gradle.api.tasks.Classpath;
+            import org.gradle.api.tasks.TaskAction;
+            import org.gradle.workers.WorkerExecutor;
+            import org.gradle.workers.WorkAction;
+            import org.gradle.workers.WorkParameters;
+            import org.gradle.workers.WorkQueue;
+
+            import javax.inject.Inject;
+
+            public class WorkerRunningPlugin implements Plugin<Project> {
+                public void apply(Project project) {
+                    ConfigurationContainer configurations = project.getConfigurations();
+                    NamedDomainObjectProvider<Configuration> bucket = configurations.register("bucket", config -> {
+                        config.setCanBeConsumed(false);
+                        config.setCanBeResolved(false);
+                    });
```

Comment:
It looks like the bucket is unused?

Reply:
Yeah. The project dependency was meant to be added to the bucket